src/soundsourceproxy.cpp

```diff
@@ -433,9 +433,69 @@ QImage SoundSourceProxy::parseCoverImage() const {
     return coverImg;
 }
 
+namespace {
+
+// Keeps the TIO alive while accessing the audio data
+// of the track. The TIO must not be deleted while
+// accessing the corresponding file to avoid file
+// corruption when writing metadata while the file
+// is still in use.
+class AudioSourceProxy: public Mixxx::AudioSource {
+public:
+    AudioSourceProxy(const AudioSourceProxy&) = delete;
+    AudioSourceProxy(AudioSourceProxy&&) = delete;
+
+    static Mixxx::AudioSourcePointer create(
+            const TrackPointer& pTrack,
+            const Mixxx::AudioSourcePointer& pAudioSource) {
+        DEBUG_ASSERT(!pTrack.isNull());
+        DEBUG_ASSERT(!pAudioSource.isNull());
+        return Mixxx::AudioSourcePointer(
+                new AudioSourceProxy(pTrack, pAudioSource));
+    }
+
+    AudioSourceProxy(
```

Comment:
We don't need  to. The class is hidden inside an anonymous namespace in the 
implementation of SoundSourceProxy and it doesn't need to know which class 
is using it. Adding a "friend class SoundSourceProxy" statement does not add 
any value.

On 01/22/2016 12:02 PM, Daniel Schürmann wrote:
>
> In src/soundsourceproxy.cpp 
> <https://github.com/mixxxdj/mixxx/pull/868#discussion_r50525226>:
>
> > +// is still in use.
> > +class AudioSourceProxy: public Mixxx::AudioSource {
> > +public:
> > +    AudioSourceProxy(const AudioSourceProxy&) = delete;
> > +    AudioSourceProxy(AudioSourceProxy&&) = delete;
> > +
> > +    static Mixxx::AudioSourcePointer create(
> > +            const TrackPointer& pTrack,
> > +            const Mixxx::AudioSourcePointer& pAudioSource) {
> > +        DEBUG_ASSERT(!pTrack.isNull());
> > +        DEBUG_ASSERT(!pAudioSource.isNull());
> > +        return Mixxx::AudioSourcePointer(
> > +                new AudioSourceProxy(pTrack, pAudioSource));
> > +    }
> > +
> > +    AudioSourceProxy(
>
> Can we make this private?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/868/files#r50525226>.
>